src/Security/SessionDownloadStrategy.php

```diff
@@ -58,24 +59,43 @@ class SessionDownloadStrategy implements DownloadStrategyInterface
      * @param ContainerInterface|SessionInterface $session
      * @param int                                 $times
      */
-    public function __construct(TranslatorInterface $translator, $session, $times)
+    public function __construct(object $translator, $session, $times)
```

Comment:
```suggestion
    public function __construct(object $translator, object $session, int $times)
```

Reply:
Is it `int` BC-break? `$times` is use as 2 argument:
`LegacyTranslatorInterface::transChoice($id, $number, array $parameters = [], $domain = null, $locale = null)`
In this case someone can use string like `5`.

Reply:
I will add tests for `$times` as string.

Reply:
Also we can not add `object` becouse we change `\InvalidArgumentException` to `\TypeError` when someone put string (like in tests). IMO this case should not be possible but it is BC-break.

Reply:
IMO, we can make changes like this in a stable branch when the BC break (strictly speaking) comes from an unusable or invalid use case.